README.md

```diff
@@ -0,0 +1,153 @@
+
+4.  Add the Firebase Android configuration file to your app:
+
+*  Click Download google-services.json to obtain your Firebase Android config file (google-services.json).
+    *  You can download your [Firebase Android config file](https://support.google.com/firebase/answer/7015592) again at any time. 
+    * Make sure the config file is not appended with additional characters, like (2).
+* Move your config file into the module (app-level) directory of your app.
+
+5.  To enable Firebase products in your app, add the [google-services plugin](https://developers.google.com/android/guides/google-services-plugin) to your Gradle files. 
+*  In your root-level (project-level) Gradle file  `(build.gradle)` , add rules to include the Google Services plugin. Check that you have Google's Maven repository, as well.
+ ```javascript
+buildscript {
+
+  repositories {
+    // Check that you have the following line (if not, add it):
+    google()  // Google's Maven repository
+  }
+
+  dependencies {
+    // ...
+
+    // Add the following line:
+    classpath 'com.google.gms:google-services:4.3.2'  // Google Services plugin
+  }
+}
+
+allprojects {
+  // ...
+
+  repositories {
+    // Check that you have the following line (if not, add it):
+    google()  // Google's Maven repository
+    // ...
+  }
+}
+```
+
+* In your module (app-level) Gradle file (usually app/build.gradle), add a line to the bottom of the file.
+
+```javascript
+apply plugin: 'com.android.application'
+
+android {
+  // ...
+}
+
+// Add the following line to the bottom of the file:
+apply plugin: 'com.google.gms.google-services'  // Google Play services Gradle plugin
+
+```
+6. Add Firebase SDKs to your app
+* To your module (app-level) Gradle file `(usually app/build.gradle)`, add the dependencies for the Firebase products that you want to use in your app.
+
+ ```javascript
+dependencies {
+  // ...
+
+  // Add the Firebase SDK for Google Analytics
+  implementation 'com.google.firebase:firebase-analytics:17.2.0'
+
+  // Add the SDKs for any other Firebase products you want to use in your app
+  // For example, to use Firebase Authentication and Cloud Firestore
+  implementation 'com.google.firebase:firebase-auth:19.0.0'
+  implementation 'com.google.firebase:firebase-firestore:21.1.0'
+
+  // Getting a "Could not find" error? Make sure that you've added
+  // Google's Maven repository to your root-level build.gradle file
+}
+```
+* Sync your app to ensure that all dependencies have the necessary versions.
+* If you added Analytics, run your app to send verification to Firebase that you've successfully integrated Firebase. Otherwise, you can skip the verification step.
+* Your device logs will display the Firebase verification that initialization is complete. If you ran your app on an emulator that has network access, the Firebase console notifies you that your app connection is complete.
+

```

Comment:
7.  Gradle properties
By default, Gradle uses a file called `(gradle.properties)` in the root directory for Android Projects. This extension file is widely used in Java projects, so it is also used for Android projects. 
The file has a simple key — value data structure.
 `(KEY_1=VALUE_1)`
 `(KEY_2=VALUE_2)` 
By default, there is a properties file in Gradle Home where Gradle system lives in your environment, they are loaded automatically by Gradle, if the filename is `(gradle.properties)`, we can use these variables `(inapp/build.gradle)` just writing their names.
Below a default Gradle properties file created by Android Studio:
```javascript
# Project-wide Gradle settings.

# IDE (e.g. Android Studio) users:
# Gradle settings configured through the IDE *will override*
# any settings specified in this file.

# For more details on how to configure your build environment visit
# http://www.gradle.org/docs/current/userguide/build_environment.html

# Specifies the JVM arguments used for the daemon process.
# The setting is particularly useful for tweaking memory settings.
org.gradle.jvmargs=-Xmx1536m

# When configured, Gradle will run in incubating parallel mode.
# This option should only be used with decoupled projects. More details, visit
# http://www.gradle.org/docs/current/userguide/multi_project_builds.html#sec:decoupled_projects
# org.gradle.parallel=true
```

And the following is an example of `(app/gradle.properties)`:

```javascript
# Gradle properties for module app
API_BASE_URL="http://example.com"
DB_FILEPATH="/common/filepath/db"
API_KEY="2FA43FADFaw432dfa"
```